tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -3134,11 +3134,35 @@ void GetTensorDimsWithProtoShape(const Tensor& tensor, nvinfer1::Dims* dims) {
   }
 }
 
+template <typename Bounds, typename Input>
+inline bool IsIntegerInBounds(const Input& inp) {
+  static_assert(std::is_integral<Input>::value,
+                "This function is only implemented for integral types.");
+  // If Input is always within the range of Bounds, return true.
+  if (std::numeric_limits<Input>::lowest() >=
+          std::numeric_limits<Bounds>::lowest() &&
+      std::numeric_limits<Input>::max() <= std::numeric_limits<Bounds>::max()) {
+    return true;
+  }
+  // Otherwise, we need to check the value of the input.
+  return (inp >= static_cast<Input>(std::numeric_limits<Bounds>::lowest()) &&
```

Comment:
`static_cast<uint32>(std::numeric_limits<int32>::lowest())` will be `2147483648` so this will never be true. In general I think it's hard to implement a general safe integer comparison function, so instead I recommend to fix `Bounds` to be `int32` and check whether `inp` is negative.

Reply:
Oh good point. I've changed it to now check explicitly whether the input type is signed (and if not, only check upper bound)